x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -25,24 +35,42 @@
 import org.elasticsearch.snapshots.SnapshotId;
 import org.elasticsearch.snapshots.SnapshotInfo;
 import org.elasticsearch.snapshots.SnapshotShardFailure;
+import org.elasticsearch.snapshots.SnapshotState;
+import org.elasticsearch.xpack.ccr.Ccr;
+import org.elasticsearch.xpack.ccr.CcrLicenseChecker;
 
 import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 /**
  * This repository relies on a remote cluster for Ccr restores. It is read-only so it can only be used to
  * restore shards/indexes that exist on the remote cluster.
  */
 public class CcrRepository extends AbstractLifecycleComponent implements Repository {
 
+    private static final String SNAPSHOT_UUID = "_latest_";
     public static final String TYPE = "_ccr_";
     public static final String NAME_PREFIX = "_ccr_";
 
     private final RepositoryMetaData metadata;
-
-    public CcrRepository(RepositoryMetaData metadata, Settings settings) {
+    private final String remoteClusterAlias;
+    private final Client client;
+    private final CcrLicenseChecker ccrLicenseChecker;
+    // The snapshot will be compatible because remote cluster connections must be compatible with our cluster
+    // version. So the snapshot version is not too important.
+    // TODO: Evaluate where we eventually want to pull the version from
+    private final Version version = Version.CURRENT;
+
+    public CcrRepository(RepositoryMetaData metadata, Client client, CcrLicenseChecker ccrLicenseChecker, Settings settings) {
         super(settings);
         this.metadata = metadata;
+        this.remoteClusterAlias = Strings.split(metadata.name(), NAME_PREFIX)[1];
```

Comment:
maybe assert one line before this that `metadata.name().startsWith(NAME_PREFIX)`?

Reply:
Done